test_runner/regress/test_compaction.py

```diff
@@ -190,3 +191,54 @@ def test_sharding_compaction(
 
     # Assert that everything is still readable
     workload.validate()
+
+
+class CompactionAlgorithm(str, enum.Enum):
+    LEGACY = "Legacy"
+    TIERED = "Tiered"
+
+
+@pytest.mark.parametrize(
+    "compaction_algorithm", [CompactionAlgorithm.LEGACY, CompactionAlgorithm.TIERED]
+)
+def test_uploads_and_deletions(
+    neon_env_builder: NeonEnvBuilder,
+    compaction_algorithm: CompactionAlgorithm,
+):
+    """
+    :param compaction_algorithm: the compaction algorithm to use.
+    """
+
+    tenant_conf = {
+        # small checkpointing and compaction targets to ensure we generate many upload operations
+        "checkpoint_distance": f"{128 * 1024}",
+        "compaction_threshold": "1",
+        "compaction_target_size": f"{128 * 1024}",
+        # no PITR horizon, we specify the horizon when we request on-demand GC
+        "pitr_interval": "0s",
+        # disable background compaction and GC. We invoke it manually when we want it to happen.
+        "gc_period": "0s",
+        "compaction_period": "0s",
+        # create image layers eagerly, so that GC can remove some layers
+        "image_creation_threshold": "1",
+        "image_layer_creation_check_threshold": "0",
+        "compaction_algorithm": json.dumps({"kind": compaction_algorithm.value}),
+    }
+    env = neon_env_builder.init_start(initial_tenant_conf=tenant_conf)
+
+    # TODO remove these allowed errors
+    # https://github.com/neondatabase/neon/issues/7707
+    # https://github.com/neondatabase/neon/issues/7759
+    env.pageserver.allowed_errors.extend(
+        [
+            ".*duplicated L1 layer.*",
+            ".*delta layer created with.*duplicate values.*",
+            ".*assertion failed: self.lsn_range.start <= lsn.*",
+            ".*HTTP request handler task panicked: task.*panicked.*",
+        ]
+    )
+
+    generate_uploads_and_deletions(env, pageserver=env.pageserver)
+
+    # We would like to assert for "duplicated L1 layer" here, but there is other errors that
+    # might occur flakily. Therefore, don't assert anything.
```

Comment:
I don't get this, I thought at least one of the `allowed_errors` above reproduces reliably?

Reply:
Originally I've thought that as well, but it turns out that the "duplicated L1 layer" error does not occur reliably. If the "assertion failed: self.lsn_range.start <= lsn" error occurs, the "duplicated L1 layer" is not reached.

see [this](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-7758/9087779519/index.html#suites/f08716c3eb261a35d11a79ed9535ded6/17a1b1c9b4e5e98d) as an example.

I have added some code that iterates over all the allowed errors we are adding and ensures that at least one of them is in the log.